docs/dev/cloud-provider-integration.md

```diff
@@ -0,0 +1,293 @@
+     # on a Node at some moment of cluster installation
+     - effect: NoSchedule
+       key: node-role.kubernetes.io/master
+       operator: Exists
+     - effect: NoExecute
+       key: node.kubernetes.io/unreachable
+       operator: Exists
+       tolerationSeconds: 120
+     # This taint is strictly required in order for CCM to be scheduled
+     # on the master Node at the moment when the cluster
+     # lacks CCM replicas to untaint the Node
+     - effect: NoSchedule
+       key: node.cloudprovider.kubernetes.io/uninitialized
+       value: "true"
+```
+
+
+Our operator is responsible for synchronization of `cloud-config` ConfigMap from `openshift-config` and `openshift-config-managed` namespace to the namespace where the CCM resources are provisioned.  The ConfigMap is named `cloud-conf `and could be mounted into a CCM pod for later use if your cloud provider requires it.
+
+Credentials secret serving to `openshift-cloud-controller-manager` namespace is carried by [https://github.com/openshift/cloud-credential-operator](https://github.com/openshift/cloud-credential-operator) for us. You need to implement your cloud provider support there, and add a `CredentialsRequest` resource in `manifests` directory. 
+
+
+## Cloud-provider fork on openshift side
+
+You are required to create your cloud-provider fork under openshift organization. This fork will be responsible for building and resolving your provider images, as well as following openshift release branching cadence.That repository has to be added into CI system and will run post submit and periodic jobs with e2e tests on your cloud-provider.
+
+You will require repository addition to [openshift/release](https://github.com/openshift/release). Here is an [PR](https://github.com/openshift/release/pull/17199) for AWS [openshift/cloud-provider-aws](https://github.com/openshift/cloud-provider-aws) repository, establishing initial set of CI jobs, including `images` job responsible for building and tagging `registry.ci.openshift.org/openshift:aws-cloud-controller-manager`
+
+Once an image is added to the CI system, it will be present in CI builds but not in release builds. You have to follow ART guidelines and create tickets to add your cloud-provider repository into release. See [https://docs.ci.openshift.org/docs/how-tos/onboarding-a-new-component/#product-builds-and-becoming-part-of-an-openshift-release-image](https://docs.ci.openshift.org/docs/how-tos/onboarding-a-new-component/#product-builds-and-becoming-part-of-an-openshift-release-image) You need openshift engineering help to set this up.
+
+It is very important to get this done step before the next one,an addition to image-references could be done only once the ART request has been resolved, or it will break nightly releases. We currently track relevant requests in this [doc](https://docs.google.com/document/d/1XB_-I9wTBwsGw-a7rwiCmOlnxTVnUdbw5odkyLKwp1k/edit?usp=sharing).
```

Comment:
```suggestion
It is very important to get this step done before the next one,an addition to image-references could be done only once the ART request has been resolved, or it will break nightly releases. We currently track relevant requests in this [doc](https://docs.google.com/document/d/1XB_-I9wTBwsGw-a7rwiCmOlnxTVnUdbw5odkyLKwp1k/edit?usp=sharing).
```